---
identifier: "CVE-2021-29534"
package_slug: "maven/org.tensorflow/parentpom"
title: "Improper Check for Unusual or Exceptional Conditions"
description: "TensorFlow is an end-to-end open source platform for machine learning.
  An attacker can trigger a denial of service via a `CHECK`-fail in `tf.raw_ops.SparseConcat`.
  This is because the implementation(https://github.com/tensorflow/tensorflow/blob/b432a38fe0e1b4b904a6c222cbce794c39703e87/tensorflow/core/kernels/sparse_concat_op.cc#L76)
  takes the values specified in `shapes[0]` as dimensions for the output shape. The
  `TensorShape` constructor(https://github.com/tensorflow/tensorflow/blob/6f9896890c4c703ae0a0845394086e2e1e523299/tensorflow/core/framework/tensor_shape.cc#L183-L188)
  uses a `CHECK` operation which triggers when `InitDims`(https://github.com/tensorflow/tensorflow/blob/6f9896890c4c703ae0a0845394086e2e1e523299/tensorflow/core/framework/tensor_shape.cc#L212-L296)
  returns a non-OK status. This is a legacy implementation of the constructor and
  operations should use `BuildTensorShapeBase` or `AddDimWithStatus` to prevent `CHECK`-failures
  in the presence of overflows. The fix will be included in TensorFlow We will also
  cherrypick this commit on TensorFlow, TensorFlow, TensorFlow and TensorFlow, as
  these are also affected and still in supported range."
date: "2021-05-17"
pubdate: "2021-05-14"
affected_range: "(,2.1.4],[2.2.0,2.2.3],[2.3.0,2.3.3],[2.4.0,2.4.2]"
fixed_versions: []
affected_versions: "All versions up to 2.1.4, all versions starting from 2.2.0 up
  to 2.2.3, all versions starting from 2.3.0 up to 2.3.3, all versions starting from
  2.4.0 up to 2.4.2"
not_impacted: ""
solution: "Unfortunately, there is no solution available yet."
urls:
- "https://nvd.nist.gov/vuln/detail/CVE-2021-29534"
cvss_v2: "AV:L/AC:L/Au:N/C:N/I:N/A:P"
cvss_v3: "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H"
uuid: "02ce21c5-8d09-4e23-860b-2ca2421b7e63"
